### PR TITLE
layers: Move all Action commands to CoreChecks

### PR DESCRIPTION
**Pull request by spencer-lunarg**
closes https://github.com/KhronosGroup/Vulkan-ValidationLayers/issues/6368

This adds a `GetObjectList` to the command buffer state object. The goal is for "Action" commands (draw, dispatch, traceRays) that we actually mark the `VkPipeline`/`VkShaderEXT` it came from

This PR moves all the checks from stateless to `CoreCheck`, it also organizes all the calls to be unified in the order of things

### CODE REVIEW

File: layers/state_tracker/cmd_buffer_state.cpp [hunk 1]
```diff
@@ -1541,3 +1541,72 @@ void CMD_BUFFER_STATE::UnbindResources() {
...
+void CMD_BUFFER_STATE::GetCurrentPipelineAndDesriptorSets(VkPipelineBindPoint pipelineBindPoint, const PIPELINE_STATE **rtn_pipe,
+                                                          const std::vector<LAST_BOUND_STATE::PER_SET> **rtn_sets) const {
+    const auto lv_bind_point = ConvertToLvlBindPoint(pipelineBindPoint);
+    const auto &last_bound = lastBound[lv_bind_point];
+    if (!last_bound.IsUsing()) {
+        return;
+    }
+    *rtn_pipe = last_bound.pipeline_state;
+    *rtn_sets = &(last_bound.per_set);
+}
```
Comment (arno-lunarg) [inline layers/state_tracker/cmd_buffer_state.cpp:1612]: For consistency, I would return a `std::pair` from this function:
```suggestion
std::pair<const PIPELINE_STATE *, const std::vector<LAST_BOUND_STATE::PER_SET> *> CMD_BUFFER_STATE::GetCurrentPipelineAndDesriptorSets(VkPipelineBindPoint pipelineBindPoint) const {
    const auto lv_bind_point = ConvertToLvlBindPoint(pipelineBindPoint);
    const auto &last_bound = lastBound[lv_bind_point];
    if (!last_bound.IsUsing()) {
        return {nullptr, nullptr};
    }
	return {last_bound.pipeline_state, &last_bound.per_set};
}
```

File: tests/unit/mesh.cpp [hunk 2]
```diff
@@ -683,12 +683,17 @@ TEST_F(NegativeMesh, BasicUsageNV) {
     ASSERT_VK_SUCCESS(result);
 
     m_commandBuffer->begin();
+    m_commandBuffer->BeginRenderPass(m_renderPassBeginInfo);
 
+    m_errorMonitor->SetDesiredFailureMsg(kErrorBit, "VUID-vkCmdDrawMeshTasksIndirectNV-None-08606");
```
Comment (arno-lunarg) [inline tests/unit/mesh.cpp:688]: Just to know, why do we have those new VUIDs being emitted in the tests?

**Review by arno-lunarg — APPROVED** [review]
LGTM!
Not sure about the motivation behind moving a lot of things from stateless to core checks? Is it mandatory because of the introduced changes?

**Comment by spencer-lunarg** [comment]
> Not sure about the motivation behind moving a lot of things from stateless to core checks?

We need to be able to print the last bound `VkPipeline`/`VkShaderEXT` which we don't have with stateless

File: tests/unit/mesh.cpp (hunk 2 above)
Comment (spencer-lunarg) [reply tests/unit/mesh.cpp:688]: If we hit a VU in Stateless it stops and never goes into CoreChecks

Before, it hit the VU below this line and stopped, now we hit multiple VUs, the "real" fix is improving the tests in the long term

File: layers/state_tracker/cmd_buffer_state.cpp (hunk 1 above)
Comment (spencer-lunarg) [reply layers/state_tracker/cmd_buffer_state.cpp:1612]: I just moved this from the Header to Source file, didn't look at it, but agree, can fix it up

Comment (arno-lunarg) [reply layers/state_tracker/cmd_buffer_state.cpp:1612]: Ah I was thinking this was new!

**Pull request merged**